plugins/modules/foreman_subnet.py

```diff
@@ -221,37 +208,34 @@ def main():
         supports_check_mode=True,
     )
 
-    (entity_dict, state) = module.parse_params()
+    entity_dict = module.clean_params()
 
     module.connect()
 
-    try:
-        # Try to find the Subnet to work on
-        entity = find_subnet(module, name=entity_dict['name'], failsafe=True)
-    except Exception as e:
-        module.fail_json(msg='Failed to find entity: %s ' % e)
-
-    if 'mask' in entity_dict and 'cidr' not in entity_dict:
-        entity_dict['cidr'] = IPNetwork('%s/%s' % (entity_dict['network'], entity_dict['mask'])).prefixlen
-    elif 'mask' not in entity_dict and 'cidr' in entity_dict:
-        entity_dict['mask'] = IPNetwork('%s/%s' % (entity_dict['network'], entity_dict['cidr'])).netmask
+    # Try to find the Subnet to work on
+    entity = module.find_resource_by_name('subnets', entity_dict['name'], failsafe=True)
 
-    if 'domains' in entity_dict:
-        entity_dict['domains'] = find_domains(module, entity_dict['domains'])
+    if not module.desired_absent:
+        if 'mask' in entity_dict and 'cidr' not in entity_dict:
+            entity_dict['cidr'] = IPNetwork('%s/%s' % (entity_dict['network'], entity_dict['mask'])).prefixlen
+        elif 'mask' not in entity_dict and 'cidr' in entity_dict:
+            entity_dict['mask'] = IPNetwork('%s/%s' % (entity_dict['network'], entity_dict['cidr'])).netmask
 
-    for feature in ('dhcp_proxy', 'tftp_proxy', 'discovery_proxy', 'dns_proxy', 'remote_execution_proxies'):
-        if feature in entity_dict:
-            entity_dict[feature] = find_smart_proxy(module, entity_dict[feature])
+        if 'domains' in entity_dict:
+            entity_dict['domains'] = module.find_resources('domains', entity_dict['domains'], thin=True)
 
-    if 'organizations' in entity_dict:
-        entity_dict['organizations'] = find_organizations(module, entity_dict['organizations'])
+        # TODO should remote_execution seach for a list?
```

Comment:
@bagasse What do you think? One Problem is, it is not even in the foreman.json atm.

Reply:
I think it should, and the UI allows that. But how to do that I don't know :)